WordPress/gutenberg
# Use proper color for block styles control

## What?
I noticed that the block styles control pills do not use the proper background color ($gray-900) while doing some other testing today. We should use the correct color to resolve this inconsistency. 

## How?
- Replaces the `$gray-800` color with `$gray-900`, which is the interface "black" used throughout the UI. 
- Uses the `$gray-600` border color that is employed throughout other controls.

## Testing Instructions

1. Open a Post or Page. 
2. Insert an Image block. 
3. See the default Styles panel, with "Default" and "Rounded". The active state will now use the $gray-900 color value that is employed throughout the UI.

## Screenshots or screencast 

Before: 

<img width="642" alt="CleanShot 2022-12-20 at 11 46 07@2x" src="https://user-images.githubusercontent.com/1813435/208720554-43770994-f2a5-430d-a3db-69ad414f4bb6.png">

After: 

<img width="647" alt="CleanShot 2022-12-20 at 11 52 01@2x" src="https://user-images.githubusercontent.com/1813435/208721754-eae991ad-a754-4da4-91ba-149ca88e92d3.png">
